### ISSUE TITLE
fancontrol uses wrong sensors

### ISSUE THREAD

**Issue by adsworth** [tuxedocomputers/tuxedo-fan-control#5]
I have a BX1506, the GUI identifies it as N350DW. The GUI/daemons assumes the CPU temperature is 12°. lmsensors shows the CPU temp at around 50°. The acpitz-virtual-0 sensors is at 12°, whatever that sensor is.

lmsensors output:
<pre>
adi@corebooklet ~> sensors
coretemp-isa-0000
Adapter: ISA adapter
Package id 0:  +49.0°C  (high = +74.0°C, crit = +80.0°C)
Core 0:        +48.0°C  (high = +74.0°C, crit = +80.0°C)
Core 1:        +47.0°C  (high = +74.0°C, crit = +80.0°C)
Core 2:        +49.0°C  (high = +74.0°C, crit = +80.0°C)
Core 3:        +48.0°C  (high = +74.0°C, crit = +80.0°C)

acpitz-virtual-0
Adapter: Virtual device
temp1:        +12.0°C  (crit = +120.0°C)

iwlwifi-virtual-0
Adapter: Virtual device
temp1:            N/A  

pch_skylake-virtual-0
Adapter: Virtual device
temp1:        +40.5°C  
</pre>

log from the daemon:
<pre>
start daemon
daemonpath : /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol ./output/dist/ --daemon
2018-12-02T16:26:10.681Zdaemon worker start: 2018-12-02T16:26:10.680Z
getDmiModelName
isRunningAsElectronApp
2018-12-02T16:26:10.682Z; check fantable
2018-12-02T16:26:10.682Z; Model: 'XXALLXX' found in fan table.

2018-12-02T16:26:10.682Z; cpuMinTemp: '44
2018-12-02T16:26:10.682Z; cpuMaxTemp: '100
2018-12-02T16:26:10.682Z; gpuMinTemp: '44
2018-12-02T16:26:10.682Z; gpuMaxTemp: '100

2018-12-02T16:26:10.688Z; Current CPU Temp is 12
2018-12-02T16:26:10.688Z; Current GPU Temp is -1
check if daemon running
process_return: tuxedofancontro

2018-12-02T16:26:11.696Z; Current CPU Temp is 12
2018-12-02T16:26:11.696Z; Current GPU Temp is -1
check if daemon running
process_return: tuxedofancontro
</pre>

**Comment by tuxedochris** [comment]
Hi adsworth,

the TUXEDO fan control use directly the ec and can have different temperature as lmsensors. But the temperature of 12 Grad Celsius is too low and look wrong. Use you any software in background these access the ec (e.g. clevo-indicator or other)? When yes, please close the processes and restart the TUXEDO fan control and verfify the results.

A other option is the bios and ec version. Can you say me the versions numbers (You can find this on the first page in bios)?

Cheers
Chris

**Comment by adsworth** [comment]
Hi Chris,

I don't have any software running which accesses the EC, at least as far as I know. is there anyway I can find out?. I'm running GNOME Manjaro and I just installed normal desktop/developer apps.
The fancontrol daemon can write successfully to the EC, because after I start the daemon the fan just runs at idle since it thinks the CPU temp is 12-13°. After a while the CPU rises to +/-74°, it throttles its frequency and the notebook becomes really slow.

My BIOS version is 1.06.03RTA and the KBC/EC Firmware is 1.06.01

Is there anyway I can probe the EC to find out the correct port (?) for the CPU temp?

Cheers
   Adi

**Comment by adsworth** [comment]
Hi Chris,

is there anyway I can help out on debugging this? 

Cheers
   Adi

**Comment by adsworth** [comment]
Hi Chris,

is there anyway I can help out on debugging this? 

Cheers
   Adi

**Comment by tuxedochris** [comment]
Hi Adi,

we have implement a new logic for the fan controlling. Can you test the new version with you device?

Cheers
Chris

**Comment by adsworth** [comment]
Hi,

I just tested the new version. It now registers the correct temperature but it doesn't adjust the fan duty.
Just now I had a CPU temperature of 76° with a fan duty of 15%. The GUI say reports "Fan Table found".

Cheers
   Adi

**Comment by tuxedochris** [comment]
Hi,

can you send the Log files? You can found it under /var/log/tuxedo-fan/

Cheers
Chris

**Comment by adsworth** [comment]
you can get the logs from here: https://oc.adi.io/s/RyAZ7rjLYJTsLAs for the next 2 days.

**Comment by tuxedochris** [comment]
Hi,

thanks for the log files. The daemon was stopped. Start the daemon and the then should be controlling the fan.

Cheers
Chris

**Comment by adsworth** [comment]
I pressed start daemon in the GUI. Maybe there is something wrong with the service file?

**Comment by tuxedochris** [comment]
Yes, that can be a reason. Uninstall the TUXEDO fan control and install the newest version from github.

**Comment by adsworth** [comment]
I rebuild from sources and I'm running it directly from the output/build/linux-unpacked directory.

This is my service file.
```
[Unit]
Description=TUXEDO Fan Control

[Service]
Type=forking
ExecStart=/bin/bash -c "Xvfb :99 & export DISPLAY=:99 && /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol --startdaemon"
ExecStop=/bin/bash -c "Xvfb :99 & export DISPLAY=:99 && /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol --stopdaemon"
ExecReload=/bin/bash -c "Xvfb :99 & export DISPLAY=:99 && /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol --restartdaemon"

[Install]
WantedBy=multi-user.target
```

and 
```
adi@corebooklet /m/d/d/N/j/tuxedo> ll /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol
-rwxr-xr-x 1 adi adi 78M 14. Jan 11:44 /mnt/datadisk/data/projects/tuxedo-fan-control/output/build/linux-unpacked/tuxedofancontrol*
adi@corebooklet /m/d/d/N/j/tuxedo> 
```

**Comment by adsworth** [comment]
Hi Chris,

any idea when there will be an new commit I can try?

Thanks
   Adi

**Comment by tuxedochris** [comment]
Hi Adi,

sorry for the late respone. The config file look good. Can you try the following
1. Start the daemon over systemctl -> systemctl start tuxedofancontrol
2. tail -f /var/log/tuxedo-fan/tuxedo-fan.log

Can you see, if the daemon write into the log file? When yes, the daemon is running.

Cheers
Chris

**Comment by adsworth** [comment]
Hi,

OK, I just did the following:

1. start the GUI as root. The GUI shows CPU Temp: 50 and CPU FAN Duty (in %): 60
2. run 'tail sudo tail -f /var/log/tuxedo-fan/tuxedo-fan.log' and 'sudo tail -f /var/log/tuxedo-fan/tuxedo-fan-daemon.log'
3. start the daemon via systemctl start tuxedofancontrol.

What happens is that:

- the fan duty drops to 12% "CPU FAN Duty (in %): 12"
- in /var/log/tuxedo-fan/tuxedo-fan-daemon.log the following two lines appear once a second:
    check if daemon running
    process_return: tuxedofancontro
   before starting the daemon only the first line would appear once a second
- in /var/log/tuxedo-fan/tuxedo-fan.log the line "Configure daemonRead fan table" appears.

I then stopped the daemon with "systemctl stop tuxedofancontrol". immediatly the FAN Duty increases to ~60%.

I hope this helps. Let me know if I can do anything else to help.

Cheers
   Adi
